subsys/bluetooth/mesh/Kconfig

```diff
@@ -103,6 +112,58 @@ menuconfig BT_MESH_ADV_EXT
 
 if BT_MESH_ADV_EXT
 
+choice BT_MESH_WORKQ_CONTEXT
+	prompt "Mesh Thread Selection"
+	default BT_MESH_WORKQ_MESH
+	help
+	  Selects in which context the BT Mesh messages will be sending.
+
+config BT_MESH_WORKQ_MESH
+	bool "Sending for BT Mesh messages in the mesh-specific work queue"
+	help
+	  When this option is selected, the Mesh will sending for Mesh messages
+	  in the mesh-specific work queue. This will ensure Mesh messages always sent.
+	  The application needs to ensure the mesh-specific work queue size is large
+	  enough, refer to BT_MESH_ADV_STACK_SIZE for the recommended minimum.
```

Comment:
```suggestion
	bool "Send messages from in the mesh-specific workqueue"
	help
	  When this option is selected, the mesh sends messages from the
	  mesh-specific workqueue. This will ensure that messages are always sent.

	  The application needs to ensure the mesh-specific workqueue size is large
	  enough. Refer to BT_MESH_ADV_STACK_SIZE for the recommended minimum.
```